runelite-client/src/main/java/net/runelite/client/plugins/suppliestracker/SuppliesTrackerPanel.java

```diff
@@ -97,13 +100,21 @@
 		overallPanel.add(overallIcon, BorderLayout.WEST);
 		overallPanel.add(overallInfo, BorderLayout.CENTER);
 
+		//Sorts boxes into usage types
 		for (ItemType type : ItemType.values())
 		{
 			SuppliesBox newBox = new SuppliesBox(itemManager, type.getLabel(), plugin, this, type);
 			logsContainer.add(newBox);
 			boxList.add(newBox);
 		}
 
+		chargesBox = new ChargesBox(itemManager, "Charges", plugin, this, ItemType.CHARGES);
```

Comment:
Do you really need to handle these 2 seperately? Because of the abstract box class, you could change `boxList`'s type to Box, and add these 2 in there as well (they got the same methods anyway!)

Reply:
Not sure the best way resolve this. All the items that go into the itemtype sorting share a tooltip method while the Charges and jewellery box use different methods. I could consolidate all the boxes into one class and have the tooltip built based on itemtype check. Would that be preferred to separate box classes?

Reply:
Removed jewellery and charges boxes. Added the code into a type check in supplies box. removed the abstract box since there is only one box now. Notated and cleaned for readability.

Reply:
I actually meant it "the other way around". If you define `abstract String buildToolTip()` in the abstract box class the implementation logic for each box can go in it's own class, while still exposing a way to get the tooltip from it.

Depending on the size of the implementation classes, having them all in a single source file could work nicely. Like

````
abstract class Box
{
	protected box() {
		// logic consistent with every implementation
	}

	abstract String buildTooltip();

	public static Box of(ItemType type) {
		// return the appropriate class here
	}

	private static class JewelleryBox extends Box
	{
		private JewelleryBox() {}

		final String buildTooltip() {
			// implementation
		}
	}
}

Reply:
So revert back to previous version except have all sources in one class instead of class for each box?